app/kumactl/cmd/uninstall/uninstall_ebpf.go

```diff
@@ -0,0 +1,312 @@
+			if err != nil {
+				return errors.Wrap(err, "Could not create Kubernetes client")
+			}
+
+			ctx, cancel := context.WithTimeout(cmd.Context(), args.Timeout)
+			defer cancel()
+
+			nodes, err := k8sClient.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+			if err != nil {
+				return errors.Wrap(err, "Failed obtaining nodes from Kubernetes cluster")
+			}
+
+			jobResource := CleanupJob{
+				jobClient:   k8sClient.BatchV1().Jobs(args.Namespace),
+				podClient:   k8sClient.CoreV1().Pods(args.Namespace),
+				startedJobs: map[string]*CleanupJobProps{},
+				stdout:      cmd.OutOrStdout(),
+				stderr:      cmd.ErrOrStderr(),
+				RWMutex:     &sync.RWMutex{},
+			}
+
+			if args.RemoveOnly {
+				if err := jobResource.Cleanup(ctx); err != nil {
+					return errors.Wrap(err, "Failed cleaning jobs")
+				}
+
+				return nil
+			}
+
+			for id, node := range nodes.Items {
+				jobName := fmt.Sprintf("%s-%d", BpfCleanupJobName, id)
+				jobSpec := genJobSpec(jobName, node.Name, &args)
+
+				if _, err := jobResource.jobClient.Create(ctx, jobSpec, metav1.CreateOptions{}); err != nil {
+					return errors.Wrap(err, "failed creating cleanup job")
+				}
+
+				jobResource.startedJobs[jobName] = &CleanupJobProps{
+					nodeName: node.Name,
+				}
+			}
+
+			watcher, err := jobResource.podClient.Watch(ctx, metav1.ListOptions{
+				LabelSelector: KumaBpfLabelSelector,
+				Watch:         true,
+			})
+			if err != nil {
+				return errors.Wrap(err, "failed to create pod watcher")
+			}
+
+			errCh := make(chan error, 1)
+
+			defer func() {
+				if e := jobResource.Cleanup(ctx); e != nil {
+					errCh <- e
+				}
+			}()
+
+			go func() {
+				errCh <- jobResource.Watch(ctx, watcher)
+			}()
+
+			select {
+			case <-ctx.Done():
+				return nil
+			case err := <-errCh:
+				return err
+			}
+		},
+	}
+
+	cmd.Flags().StringVar(&args.Namespace, "namespace", args.Namespace, "namespace where job is created")
+	cmd.Flags().StringVar(&args.BPFFsPath, "bpffs-path", args.BPFFsPath, "path where bpf programs were installed")
+	cmd.Flags().DurationVar(&args.Timeout, "timeout", args.Timeout, "timeout for whole process of removing left files")
+	cmd.Flags().StringVar(&args.CleanupImageVersion, "cleanup-image-version", args.CleanupImageVersion, "version of cleanup ebpf job image")
+	cmd.Flags().BoolVar(&args.RemoveOnly, "remove-only", args.RemoveOnly, "cleanup jobs and pods only")
+
+	return cmd
+}
+
+func genJobSpec(jobName, nodeName string, args *ebpfArgs) *batchv1.Job {
+	return &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: jobName,
+		},
+		Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						AppLabel: BpfCleanupJobName,
+					},
+				},
+				Spec: corev1.PodSpec{
+					NodeName: nodeName,
+					Containers: []corev1.Container{
+						{
+							Name:  BpfCleanupJobName,
+							Image: fmt.Sprintf("%s:%s", BpfCleanupImage, args.CleanupImageVersion),
```

Comment:
You need to make this configurable at least for KM but also because some users build their own images

Reply:
ack

Reply:
addressed